### ISSUE TITLE
Plugin Support

### ISSUE THREAD

**Issue by tomtom87** [keystonejs/keystone#4713]
I love the keystone project and would like to start writing some plugins for keystone so it can help grow the ecosystem and gain more adoption. 

Is there any guidelines or way you would like this to be approached? I'd rather get some feedback from the main devs here before starting this. I read on a few issues that plugins are planned, would very much like to help getting the wheels rolling.

Cheers

**Comment by tomtom87** [comment]
Seeing how simple it is to just add NPM packages into Keystone, possibly we could just have a basic framework to adhere too and aggregate the plugins parameters to the admin panel. Doesn't need to be too far fetched just a simple form to change the plugins settings.

I could get the ball rolling by adding in a open graph plugin as an example possibly

**Comment by kennedybaird** [comment]
I'm looking at doing this as well.

@tomtom87 Did you look into this at all, integrating with the Admin UI?

**Comment by tomtom87** [comment]
@kennedybaird I am starting on it in this next quarter. I have managed to add in several packages so far without any need for this admin panel so was thinking maybe is possibly just overkill as most users of Keystone are developer literate. 

That being said, going forward it would be nice to have some kind of format I could develop keystone specific plugins for, that could have a modular admin UI for example to setup SEO meta information without needing to hack strings into the models and templates.

If I make any modifications that are good enough for release will make a PR. Still thinking about it quiet often

**Comment by kennedybaird** [comment]
Great, I am interested as I'm looking at developing some things for a
project that have reasonably generic applications - so it'll be great to
see what you come up with.
